### ISSUE TITLE
Add HA support.

### ISSUE THREAD

**Issue by blmhemu** [mholt/caddy-dynamicdns#48]
**Example Scenario**
Two caddy instances on different hosts with ips 11.11.11.11 and 11.11.11.12 . Need to insert DNS records for both the IPs (for dns load balancing).

**Notes**
I understand I might be opening a can of worms (of multi instance coordination, distributed systems etc.). But thought of giving it a shot.

**Comment by mholt** [comment]
Hoo boy.

Current design of this plugin is only for simple deployments.

We could discuss a design that enables use in HA environments though.

**Comment by blmhemu** [comment]
I was thinking along the lines of serf + keepalived
Serf: To infect membership details.
Keepalived (or a version of keepalived) to select the "active" instance (in say a 3 member HA deployment) which can check for dns entries and place them.

I realize, I might be treading in the territory of consul (serf + raft).

**Comment by mholt** [comment]
Yeah, that seems like a lot of work. Forking the project to implement this seems like a good approach, and then if you like the implementation you can make a PR and we can decide whether it's a good fit to be merged in.
